packages/stream_chat_flutter/lib/src/message_input/audio_message_send_button/stream_audio_message_controllers.dart

```diff
@@ -0,0 +1,186 @@
+import 'dart:async';
+import 'dart:math';
+
+import 'package:flutter/material.dart';
+import 'package:flutter/widgets.dart';
+import 'package:stream_chat_flutter/src/message_input/audio_message_send_button/stream_audio_message_gesture_state.dart';
+import 'package:stream_chat_flutter/stream_chat_flutter.dart';
+
+/// Widget that displays the audio recording controls.
+class StreamAudioMessageControllers extends StatefulWidget {
+  /// Creates a new StreamAudioMessageControllers
+  const StreamAudioMessageControllers({super.key});
+
+  @override
+  State<StreamAudioMessageControllers> createState() =>
+      _StreamAudioMessageControllersState();
+}
+
+class _StreamAudioMessageControllersState
+    extends State<StreamAudioMessageControllers> with TickerProviderStateMixin {
+  Color? iconColor;
+  DateTime? _startTime;
+  Timer? _timer;
+  late final AnimationController _controller;
+
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+    final audioRecordingMessageTheme = AudioRecordingMessageTheme.of(context);
+
+    _controller = AnimationController(
+      duration: const Duration(milliseconds: 500),
+      vsync: this,
+    );
+    _controller.forward();
+
+    Timer.periodic(
+      const Duration(seconds: 1),
+      (timer) {
+        timer.cancel();
+        if (mounted) {
+          setState(() {
+            iconColor =
+                audioRecordingMessageTheme.recordingIndicatorColorActive;
+            _startTime = DateTime.now();
+          });
+        }
+      },
+    );
+
+    _timer = Timer.periodic(
+      const Duration(seconds: 1),
+      (timer) {
+        if (_startTime == null) {
+          return;
+        }
+        setState(() {});
+      },
+    );
```

Comment:
If this is for setting clock count, can we use a `ValueNotifier` coupled with a builder for it instead of a full `setState()` as it is a stateful widget anyway?

Reply:
This will change with the implementation of the recording function, I will revisit it then